### ISSUE TITLE
Update nginx-proxy for M1 Mac Support

### ISSUE THREAD

**Issue by jmjones88** [10up/phpfpm-image#21]
**Describe the bug**
When the `nginx-proxy` container starts on an M1 Mac, you get the following error:
```
runtime: failed to create new OS thread (have 2 already; errno=22)
fatal error: newosproc
```

**Steps to Reproduce**
1. Start the global containers
2. Look at the Error in Docker Desktop

**Expected behavior**
The container should start without issue

**Environment information**
 - Device: MacBook Air (M1, 2020)
 - OS: MacOS 11.2.3

**Additional context**
The issue has been solved temporarily on the main repo https://github.com/nginx-proxy/nginx-proxy/issues/1550

**Comment by jeremiahbratton** [comment]
I am having the same issue, and as I do some research on my end on how to get things working I found this issue. I thought it might be helpful to note that there is a conversation that relates to this happening in an issue for wp-local-docker-v2

For anyone searching for a fix like me this might help close the loop for existing knowledge.

Issue Link: https://github.com/10up/wp-local-docker-v2/issues/188

**Comment by dustinrue** [comment]
Hi everyone. I have updated and rebuilt the proxy so that it supports linux/amd64 and linux/arm64. This should provide support for the M1 chip. It would be great if someone could test this. Here is what you need to do to test:

* Issue `10updocker stop all` and make sure _all_ docker containers that are part of wp-local-docker global servers are destroyed (check `docker ps -a`)
* Edit the `~/.wplocaldocker/global/docker-compose.yml` file and update the image entry for the gateway from `10up/nginx-proxy` to `10up/nginx-proxy:multiarch`
* Start an environment and confirm this updated image works as expected

I need to see this tested on both an x86 system as well as an arm64 (M1) system.

Thank you!

**Comment by jeremiahbratton** [comment]
@dustinrue updated and made the change. The gateway is still having trouble after a 10updocker start. I am on an M1 Mac Mini

```
Creating global_dns_1 ... 
Creating global_mysql_1 ... 
Creating global_gateway_1 ... 
Creating global_mailcatcher_1 ... 
Host is already in use by another container
Creating global_gateway_1     ... error

ERROR: for global_gateway_1  Cannot start service gateway: driver failed programming external connectivity on endpoint global_gateway_1 (d0a9772620d0babb49af60479c6abc59277766db553d67b7a208535d7520b7a0): Bind for 127.0.0.1:443 failed: port is already allocated
Creating global_mailcatcher_1 ... done
Creating global_mysql_1       ... done
Creating global_phpmyadmin_1  ... 
Creating global_dns_1         ... done
Creating global_phpmyadmin_1  ... done

ERROR: for gateway  Cannot start service gateway: driver failed programming external connectivity on endpoint global_gateway_1 (d0a9772620d0babb49af60479c6abc59277766db553d67b7a208535d7520b7a0): Bind for 127.0.0.1:443 failed: port is already allocated
Encountered errors while bringing up the project.
```

and from the logs in docker
`global_gateway_1 exited with code 128` and `Error invoking remote method 'docker-start-container': Error: (HTTP code 500) server error - driver failed programming external connectivity on endpoint global_gateway_1 (95e4d50c8991d11d52ca3ac83f10d58191843066af21045a5e5227676183944f): Bind for 127.0.0.1:443 failed: port is already allocated`

**Comment by dustinrue** [comment]
Make sure you don't have something using that port already. Docker containers, some other local dev system

**Comment by jeremiahbratton** [comment]
@dustinrue That I can see there is nothing else that would be running and using that port. No other containers are running. I am a little out of my depth with things like this, it is possible I am not looking at the right thing. Do you have any steps you would perform that could help me get more information for you?

**Comment by dustinrue** [comment]
Make sure you do `10updocker stop all` and then `docker ps`. It should come back with nothing. If it doesn't...you'll have to resolve that. If it is empty you can try `telnet localhost 443` to see if something responds. If something responds (it looks like this):

```
telnet localhost 443
Trying ::1...
telnet: connect to address ::1: Connection refused
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
```

Then _something_ has that port open. You can determine using this command

`sudo lsof -i -P -n  | grep -i listen`

**Comment by jeremiahbratton** [comment]
@dustinrue Thank you for that... ok, found it. Something must have gone wrong during the first wind up after updating. Going into docker desktop and deleting the `10up/nginx-proxy:multiarch` container then doing another `10updocker start` got it running. The mySQL container is still failing so I can't fully confirm the setup. But I can see a green light for `10up/nginx-proxy:multiarch` in docker desktop

Also, I can at least get a 503 at the project URL so that is definitely progress.

**Comment by dustinrue** [comment]
Yup. The MySQL container needs to be swapped out as well and I’ll see what I can do about it in the coming days. Rumor has it you can use MariaDB but this is untested

**Comment by jeremiahbratton** [comment]
I was given that recommendation in the other thread (https://github.com/10up/wp-local-docker-v2/issues/188) but doesn't seem to be as simple as just changing mysql:5 to mariadb but I digress... Thank you for your work and time.

**Comment by jeremiahbratton** [comment]
Update: I guess it was as simple as changing mysql:5 to mariadb in docker-compose.yml

I can now confirm for you that after the update, and the tweak to the mysql image I can access the local environment.

**Comment by azatecas** [comment]
@dustinrue thanks it now works great and i can be on the same workflow as the rest of my team!
